app/addons/documents/routes-index-editor.js

```diff
@@ -28,40 +28,53 @@ define([
   'addons/documents/pagination/pagination.react',
   'addons/documents/header/header.react',
   'addons/documents/header/header.actions',
+  'addons/documents/sidebar/actions'
 ],
 
 function (app, FauxtonAPI, Helpers, BaseRoute, Documents, IndexEditorComponents, ActionsIndexEditor,
           Databases, Components, IndexResultsStores, IndexResultsActions,
-          IndexResultsComponents, ReactPagination, ReactHeader, ReactHeaderActions) {
+          IndexResultsComponents, ReactPagination, ReactHeader, ReactHeaderActions, SidebarActions) {
 
 
   var IndexEditorAndResults = BaseRoute.extend({
-    layout: 'two_pane',
+    layout: 'with_tabs_sidebar',
     routes: {
-      'database/:database/new_view': 'newViewEditor',
-      'database/:database/new_view/:designDoc': 'newViewEditor',
+      'database/:database/new_view': {
+        route: 'createView',
+        roles: ['fx_loggedIn']
+      },
+      'database/:database/new_view/:designDoc': {
+        route: 'createView',
+        roles: ['fx_loggedIn']
+      },
       'database/:database/_design/:ddoc/_view/:view': {
-        route: 'viewFn',
+        route: 'showView',
+        roles: ['fx_loggedIn']
+      },
+      'database/:database/_design/:ddoc/_view/:view/edit': {
```

Comment:
Hmm, good point. How do you suggest handling this? The edit page are separate. I guess I can keep the same URL for both, and just handle routing exclusively in the JS... but having a way to link directly to it through a URL string alone is kinda convenient. (Query string param...? Seems klutzy).